sdk/src/azure/platform/az_curl.c

```diff
@@ -58,7 +59,8 @@ static AZ_NODISCARD az_result _az_http_client_curl_code_to_result(CURLcode code)
 }
 
 // returning AZ error on CURL Error
-#define AZ_RETURN_IF_CURL_FAILED(exp) AZ_RETURN_IF_FAILED(_az_http_client_curl_code_to_result(exp))
+#define _az_RETURN_IF_CURL_FAILED(exp) \
```

Comment:
Question: Should this be in a do/while loop?

Reply:
No. It is the multi-statement macros "returning void" that should be. `AZ_RETURN_IF_FAILED()` would wrap it in do-while.